plugins/web/opentelemetry-instrumentation-user-interaction/src/instrumentation.ts

```diff
@@ -464,41 +489,54 @@ export class UserInteractionInstrumentation extends InstrumentationBase<unknown>
           Array.isArray(applyArgs) && applyArgs[0] instanceof Event
             ? applyArgs[0]
             : undefined;
-        const target = event?.target;
+
+        const target = plugin._allowEventType(task.eventName) && event?.target;
         let span: api.Span | undefined;
-        const activeZone = this;
+
         if (target) {
-          span = plugin._createSpan(target, task.eventName);
+          if (!plugin._currentInteractionZone) {
+            // create new span for interaction from current task zone
+            span = plugin._createSpan(target, task.eventName);
+
+            if (span) {
+              api.context.with(
+                api.trace.setSpan(api.context.active(), span),
+                () => {
+                  plugin._currentInteractionZone = Zone.current;
+                }
+              );
+
+              // Reset the current interaction zone after 50ms
+              // This allows new clicks to create spans when the previous interaction is still not complete
+              const spanData = plugin._spansData.get(span);
+              if (spanData) {
+                Zone.root.run(() => {
```

Comment:
Same here. Please leave a comment why this needs to be run in the root zone

Reply:
This is the same reasoning as above, but I'll make sure to mark all locations where the plugin registers an async task since by rule these run in the same execution context. Without doing this you're going to get incorrect taskCount state values.